core/src/main/java/org/apache/accumulo/core/file/rfile/RFile.java

```diff
@@ -1015,14 +1018,23 @@ public void init(SortedKeyValueIterator<Key,Value> source, Map<String,String> op
 
     }
 
+    /**
+     * @return map of locality group names to column families. The default locality group will have {@code null} for a name. RFile will only track up to
+     *         {@value Writer#MAX_CF_IN_DLG} families for the default locality group. After this it will stop tracking. For the case where the default group has
+     *         more thn {@value Writer#MAX_CF_IN_DLG} families an empty list of families is returned.
+     * @see LocalityGroupUtil#seek(Reader, Range, String, Map)
+     */
     public Map<String,ArrayList<ByteSequence>> getLocalityGroupCF() {
       Map<String,ArrayList<ByteSequence>> cf = new HashMap<>();
 
       for (LocalityGroupMetadata lcg : localityGroups) {
-        ArrayList<ByteSequence> setCF = new ArrayList<>();
+        ArrayList<ByteSequence> setCF;
 
-        for (Entry<ByteSequence,MutableLong> entry : lcg.columnFamilies.entrySet()) {
-          setCF.add(entry.getKey());
+        if (lcg.columnFamilies == null) {
+          Preconditions.checkState(lcg.isDefaultLG, " Group %s has null families. Only expect default locality group to have null families.", lcg.name);
```

Comment:
Is there any other useful information that could be provided? For example, would it make sense to print out the rest of the locality group column families and help identify if others are listed as a the default and have no column families?